core/hash/deconstruct_keys_spec.rb

```diff
@@ -0,0 +1,25 @@
+require_relative '../../spec_helper'
+
+ruby_version_is "2.7" do
+  describe "Hash#deconstruct_keys" do
+    it "returns self" do
+      hash = {a: 1, b: 2}
+
+      hash.deconstruct_keys([:a, :b]).should equal hash
+    end
+
+    it "requires one argument" do
+      -> {
+        {a: 1}.deconstruct_keys
+      }.should raise_error(ArgumentError, /wrong number of arguments \(given 0, expected 1\)/)
+    end
+
+    it "ignores argument" do
+      hash = {a: 1, b: 2}
+
+      hash.deconstruct_keys([:a]).should == {a: 1, b: 2}
+      hash.deconstruct_keys(0   ).should == {a: 1, b: 2}
+      hash.deconstruct_keys(''  ).should == {a: 1, b: 2}
```

Comment:
Do you know if this is intended behavior?

Reply:
@eregon 

Looks like so - https://github.com/ruby/ruby/blob/master/hash.c#L4591-L4595

Keys passed as argument are just a hint to optimize performance. `Struct#deconstruct_keys` uses keys... Hash doesn't. At least for now. Hope it will be changed in the next release.